### PR TITLE
fix(chart): curves in graph not synchronized on display

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Since Centreon 19.10.0 RRDCacheD configuration is no more global but per Centreon RRD broker configuration.

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

- On a Centreon 19.04.0 enable RRDCacheD using PR #6955 
- Update to Centreon 19.10.0
- Wait 1h
- Try to display a chart with 2 curves (traffic for example)
- Both curves should be synchronized and displayed

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [x] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/class/centreonGraph.class.php [hunk 1]
```diff
@@ -206,6 +207,22 @@ public function __construct($user_id, $index = null, $debug = 0, $compress = nul
...
+            FROM cfg_centreonbroker_info AS cbi
+            INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
+            INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
+            WHERE ns.localhost = '1'
+            AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
+        );
+        while ($row = $result->fetch()) {
+            if ($row['config_key'] == 'rrd_cached_option') {
+                $this->rrdCachedOptions['rrd_cached_option'] = $row['config_value'];
+            } elseif ($row['config_key'] == 'rrd_cached') {
```
Comment (sc979) [inline www/class/centreonGraph.class.php]:
```suggestion
            } elseif ($row['config_key'] === 'rrd_cached') {
```

File: www/class/centreonGraphNg.class.php [hunk 2]
```diff
@@ -228,6 +229,23 @@ public function __construct($userId)
...
+        $stmt = $this->db->query("SELECT config_key, config_value
+            FROM cfg_centreonbroker_info AS cbi
+            INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
+            INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
+            WHERE ns.localhost = '1'
+            AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
+        );
+
+        while ($row = $stmt->fetch()) {
+            if ($row['config_key'] == 'rrd_cached_option') {
```
Comment (sc979) [inline www/class/centreonGraphNg.class.php]:
```suggestion
            if ($row['config_key'] === 'rrd_cached_option') {
```

File: www/class/centreonGraphNg.class.php [hunk 3]
```diff
@@ -228,6 +229,23 @@ public function __construct($userId)
...
+                $this->rrdCachedOptions['rrd_cached_option'] = $row['config_value'];
+            } elseif ($row['config_key'] == 'rrd_cached') {
```
Comment (sc979) [inline www/class/centreonGraphNg.class.php]:
```suggestion
            } elseif ($row['config_key'] === 'rrd_cached') {
```

File: www/class/centreonGraphNg.class.php [hunk 4]
```diff
@@ -1231,23 +1249,22 @@ private function checkDBAvailability($metricId)
...
         $errstr = '';
-        if (isset($this->generalOpt['rrdcached_port']['value'])
-            && trim($this->generalOpt['rrdcached_port']['value']) != '') {
-            $sock = fsockopen('127.0.0.1', trim($this->generalOpt['rrdcached_port']['value']), $errno, $errstr);
-        } elseif (isset($this->generalOpt['rrdcached_unix_path']['value'])
-            && trim($this->generalOpt['rrdcached_unix_path']['value']) != '') {
-            $sock = fsockopen('unix://' . trim($this->generalOpt['rrdcached_unix_path']['value']), $errno, $errstr);
+        if ($this->rrdCachedOptions['rrd_cached_option'] == 'tcp') {
+            $sock = fsockopen('127.0.0.1', trim($this->rrdCachedOptions['rrd_cached']), $errno, $errstr);
+        } elseif ($this->rrdCachedOptions['rrd_cached_option'] == 'unix') {
```
Comment (sc979) [inline www/class/centreonGraphNg.class.php]:
```suggestion
        } elseif ($this->rrdCachedOptions['rrd_cached_option'] === 'unix') {
```

File: www/class/centreonGraph.class.php [hunk 5]
```diff
@@ -1791,20 +1808,14 @@ protected function flushRrdcached($metricsId)
...
-            $sock = @fsockopen('127.0.0.1', trim($this->generalOpt['rrdcached_port']), $errno, $errstr);
-            if ($sock === false) {
-                return false;
-            }
-        } elseif (isset($this->generalOpt['rrdcached_unix_path'])
-            && trim($this->generalOpt['rrdcached_unix_path']) != ''
-        ) {
-            $sock = @fsockopen('unix://' . trim($this->generalOpt['rrdcached_unix_path']), $errno, $errstr);
+        if ($this->rrdCachedOptions['rrd_cached_option'] == 'tcp') {
+            $sock = fsockopen('127.0.0.1', trim($this->rrdCachedOptions['rrd_cached']), $errno, $errstr);
```
[the remark this answers is not in the document]
Comment (lpinsivy) [reply www/class/centreonGraph.class.php]: yes

File: www/class/centreonGraph.class.php [hunk 6]
```diff
@@ -206,6 +207,22 @@ public function __construct($user_id, $index = null, $debug = 0, $compress = nul
...
+        /* Get RRDCacheD options */
+        $result = $this->DB->query("SELECT config_key, config_value
+            FROM cfg_centreonbroker_info AS cbi
+            INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
+            INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
+            WHERE ns.localhost = '1'
+            AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
+        );
+        while ($row = $result->fetch()) {
+            if ($row['config_key'] === 'rrd_cached_option') {
```
Comment (kduret) [inline www/class/centreonGraph.class.php]: could avoid if/elsif : 
$this->rrdCachedOptions[$row['config_key']] = $row['config_value'];

File: www/class/centreonGraph.class.php [hunk 7]
```diff
@@ -1780,8 +1797,8 @@ private function checkDBAvailability($metric_id)
      */
     protected function flushRrdcached($metricsId)
     {
-        if (!isset($this->generalOpt['rrdcached_enable'])
-            || $this->generalOpt['rrdcached_enable'] == 0
+        if (!isset($this->rrdCachedOptions['rrd_cached_option'])
+            || !preg_match('/(unix|tcp)/', $this->rrdCachedOptions['rrd_cached_option'])
```
Comment (kduret) [inline www/class/centreonGraph.class.php]:
```suggestion
            || !in_array($this->rrdCachedOptions['rrd_cached_option'], ['unix', 'tcp'])
```

File: www/class/centreonGraphNg.class.php [hunk 8]
```diff
@@ -228,6 +229,23 @@ public function __construct($userId)
...
+        $stmt = $this->db->query("SELECT config_key, config_value
+            FROM cfg_centreonbroker_info AS cbi
+            INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
+            INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
+            WHERE ns.localhost = '1'
+            AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
+        );
+
+        while ($row = $stmt->fetch()) {
+            if ($row['config_key'] === 'rrd_cached_option') {
```
Comment (kduret) [inline www/class/centreonGraphNg.class.php]: could avoid if/elsif :
$this->rrdCachedOptions[$row['config_key']] = $row['config_value'];

File: www/class/centreonGraphNg.class.php [hunk 9]
```diff
@@ -1231,23 +1249,22 @@ private function checkDBAvailability($metricId)
      */
     protected function flushRrdcached($metricsId)
     {
-        if (!isset($this->generalOpt['rrdcached_enable']['value'])
-            || $this->generalOpt['rrdcached_enable']['value'] == 0) {
+        if (!isset($this->rrdCachedOptions['rrd_cached_option'])
+            || !preg_match('/(unix|tcp)/', $this->rrdCachedOptions['rrd_cached_option'])
```
Comment (kduret) [inline www/class/centreonGraphNg.class.php]:
```suggestion
            || !in_array($this->rrdCachedOptions['rrd_cached_option'], ['unix', 'tcp'])
```

File: www/class/centreonGraphStatus.class.php [hunk 10]
```diff
@@ -133,28 +134,22 @@ public function getData()
      */
     public function flushRrdCached($indexData)
     {
-        if (false === isset($this->generalOpt['rrdcached_enabled']) ||
-            $this->generalOpt['rrdcached_enabled'] == 0
+        if (!isset($this->rrdCachedOptions['rrd_cached_option'])
+            || !preg_match('/(unix|tcp)/', $this->rrdCachedOptions['rrd_cached_option'])
```
Comment (kduret) [inline www/class/centreonGraphStatus.class.php]:
```suggestion
            || !in_array($this->rrdCachedOptions['rrd_cached_option'], ['unix', 'tcp'])
```

File: www/class/centreonGraphStatus.class.php [hunk 11]
```diff
@@ -209,6 +204,33 @@ protected function getOptions()
...
+            FROM cfg_centreonbroker_info AS cbi
+            INNER JOIN cfg_centreonbroker AS cb ON (cb.config_id = cbi.config_id)
+            INNER JOIN nagios_server AS ns ON (ns.id = cb.ns_nagios_server)
+            WHERE ns.localhost = '1'
+            AND cbi.config_key IN ('rrd_cached_option', 'rrd_cached')"
+        );
+
+        $rrdCachedOptions = [];
+        while ($row = $result->fetch()) {
+            if ($row['config_key'] === 'rrd_cached_option') {
```
Comment (kduret) [inline www/class/centreonGraphStatus.class.php]: could avoid if/elsif :
$this->rrdCachedOptions[$row['config_key']] = $row['config_value'];

**Pull request merged**